ui/src/BulkUploads.js

```diff
@@ -0,0 +1,336 @@
+
+  onProcessJob = () => {
+    fetch(`/api/job/${this.state.selectedJob}/process`, {
+      method: "POST",
+    });
+  };
+
+  onCancelJob = () => {
+    fetch(`/api/job/${this.state.selectedJob}/cancel`, {
+      method: "POST",
+    });
+  };
+
+  render() {
+    var selectedJob;
+    var detailsDialogTitle;
+    var loadPermission;
+    switch (this.state.selectedJobType) {
+      case "BULK_REFUSAL":
+        selectedJob = this.state.bulkRefusalJobs.find(
+          (job) => job.id === this.state.selectedJob
+        );
+        detailsDialogTitle = "Bulk Refusal Detail";
+        loadPermission = "LOAD_BULK_REFUSAL";
+        break;
+      default:
+    }
+
+    return (
+      <div style={{ padding: 20 }}>
+        <Link
+          to={`/collex?surveyId=${this.props.surveyId}&collexId=${this.props.collectionExerciseId}`}
+        >
+          ← Back to collection exercise details
+        </Link>
+        <Typography variant="h4" color="inherit">
+          Uploaded Bulk Process Files
+        </Typography>
+        {this.buildBulkProcessTable(
+          this.state.bulkRefusalJobs,
+          "BULK_REFUSAL",
+          "Bulk Refusals",
+          "LOAD_BULK_REFUSAL",
+          "VIEW_BULK_REFUSAL_PROGRESS"
+        )}
+        <Dialog open={this.state.uploadInProgress}>
+          <DialogContent style={{ padding: 30 }}>
+            <Typography variant="h6" color="inherit">
+              Uploading file...
+            </Typography>
+            <LinearProgress
+              variant="determinate"
+              value={this.state.fileProgress * 100}
+              style={{ marginTop: 20, marginBottom: 20, width: 400 }}
+            />
+            <Typography variant="h6" color="inherit">
+              {Math.round(this.state.fileProgress * 100)}%
+            </Typography>
+          </DialogContent>
+        </Dialog>
+        <Snackbar
+          open={this.state.fileUploadSuccess}
+          autoHideDuration={6000}
+          onClose={this.handleClose}
+          anchorOrigin={{
+            vertical: "bottom",
+            horizontal: "left",
+          }}
+        >
+          <SnackbarContent
+            style={{ backgroundColor: "#4caf50" }}
+            message={"File upload successful!"}
+          />
+        </Snackbar>
+        <JobDetails
+          jobTitle={detailsDialogTitle}
+          job={selectedJob}
+          showDetails={this.state.showDetails}
+          handleClosedDetails={this.handleClosedDetails}
+          onClickAway={this.handleClosedDetails}
+          onProcessJob={this.onProcessJob}
+          onCancelJob={this.onCancelJob}
+          authorisedActivities={this.state.authorisedActivities}
+          loadPermission={loadPermission}
+        ></JobDetails>
+      </div>
+    );
+  }
+
+  buildBulkProcessTable(
+    bulkJobs,
+    jobType,
+    jobTitle,
+    loadPermission,
+    viewerPermission
+  ) {
+    const bulkJobTableRows = bulkJobs.map((job, index) => (
+      <TableRow key={job.createdAt}>
```

Comment:
missed this... why have you used this as an identifier? `index` is the correct _unique_ identifier... can you raise a PR to fix pls

Reply:
ah I see you just copied my bad code from `SampleUpload` lol